src/libponyc/pass/expr.c

```diff
@@ -292,6 +292,11 @@ ast_result_t pass_expr(ast_t** astp, pass_opt_t* options)
     case TK_ADDRESS:    r = expr_addressof(options, ast); break;
     case TK_DIGESTOF:   r = expr_digestof(options, ast); break;
 
+    case TK_OBJECT:
```

Comment:
That's moving object literals from the sugar pass (where they used to be, as pure sugar for an anonymous type and a constructor) to the expression pass, where they live now, because they need to know about the existence and type of variables within lexical scope.